File: jdroid-android-core/src/main/java/com/jdroid/android/exception/SnackbarErrorDisplayer.java
package com.jdroid.android.exception;

import androidx.annotation.IdRes;
import androidx.annotation.StringRes;
import com.google.android.material.snackbar.Snackbar;
import androidx.fragment.app.FragmentActivity;
import android.view.View;

import com.jdroid.android.snackbar.SnackbarBuilder;

public class SnackbarErrorDisplayer extends AbstractErrorDisplayer {

	private SnackbarBuilder snackbarBuilder = new SnackbarBuilder();

	@Override
	public void onDisplayError(FragmentActivity activity, String title, String description, Throwable throwable) {
		if (activity != null) {
			snackbarBuilder.setDescription(description);
			Snackbar snackbar = snackbarBuilder.build(activity);
			snackbar.show();
		}
	}

	public void setActionTextResId(@StringRes int actionTextResId) {
		snackbarBuilder.setActionTextResId(actionTextResId);
	}

	public void setOnClickListener(View.OnClickListener onClickListener) {
		snackbarBuilder.setOnClickListener(onClickListener);
	}

	public void setParentLayoutId(@IdRes int parentLayoutId) {
		snackbarBuilder.setParentLayoutId(parentLayoutId);
	}

	public void setDuration(@Snackbar.Duration int duration) {
		snackbarBuilder.setDuration(duration);
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/exception/SnackbarErrorDisplayer.kt
package com.jdroid.android.exception

import android.view.View
import androidx.annotation.IdRes
import androidx.annotation.StringRes
import androidx.fragment.app.FragmentActivity
import com.google.android.material.snackbar.BaseTransientBottomBar
import com.jdroid.android.snackbar.SnackbarBuilder

class SnackbarErrorDisplayer : AbstractErrorDisplayer() {

    private val snackbarBuilder = SnackbarBuilder()

    public override fun onDisplayError(
        activity: FragmentActivity?,
        title: String,
        description: String,
        throwable: Throwable
    ) {
        if (activity != null) {
            snackbarBuilder.setDescription(description)
            snackbarBuilder.build(activity).show()
        }
    }

    fun setActionTextResId(@StringRes actionTextResId: Int) {
        snackbarBuilder.setActionTextResId(actionTextResId)
    }

    fun setOnClickListener(onClickListener: View.OnClickListener) {
        snackbarBuilder.setOnClickListener(onClickListener)
    }

    fun setParentLayoutId(@IdRes parentLayoutId: Int) {
        snackbarBuilder.setParentLayoutId(parentLayoutId)
    }

    fun setDuration(@BaseTransientBottomBar.Duration duration: Int) {
        snackbarBuilder.setDuration(duration)
    }
}
